src/postgres/inventory/type/PgClassType.ts

```diff
@@ -80,9 +80,21 @@ class PgClassType extends PgType<PgRow> implements ObjectType<PgRow> {
     if (typeof pgValue !== 'object')
       throw new Error(`Postgres value of object type must be an object, not '${typeof pgValue}'.`)
 
-    return new Map(Array.from(this.fields).map<[string, mixed]>(([fieldName, field]) => {
-      return [fieldName, field.type.transformPgValueIntoValue(pgValue[field.pgAttribute.name])]
-    }))
+    // Parse official pg fields
+    const obj = {};
+    Array.from(this.fields).filter<[string, mixed]>(([fieldName, field]) => pgValue.hasOwnProperty(field.pgAttribute.name)).map<[string, mixed]>(([fieldName, field]) => {
+      obj[fieldName] = field.type.transformPgValueIntoValue(pgValue[field.pgAttribute.name])
+    })
+    // Pass through other fields unchanged
+    for (const key in pgValue) {
```

Comment:
I've not checked - does field.getValue have some magic coercion in it? If so do make sure I don't lose that!